server/src/storage/commits.ts

```diff
@@ -101,13 +101,22 @@ const updateCommit = async (
         value: 1,
       });
       break;
-    case 'completed':
+    case 'completed': {
       listingUpdateRules.push({
         property: 'numCompleted',
         op: 'add',
         value: 1,
       });
+
+      const affectedListing = await getEntity(LISTING_KIND, affectedListingId);
+      if (affectedListing.numCompleted + 1 === affectedListing.numCommits) {
```

Comment:
Add an explanatory comment that after the update of numCompleted add, "numCompleted will be equal to numCommits" which makes the listing status to be completed.

Reply:
@xlai20 Okay!